woocommerce/woocommerce-gutenberg-products-block
# Storybook and TS migration of `Icon` component

This PR adds stories for the `Icon` component and migrates it to TypeScript.

Fixes #5257

### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run storybook`
2. Explore the [Icon stories](http://localhost:6006/?path=/story/woocommerce-blocks-woocommerce-icons--base-icon). You should see two stories: a base one with all the controls, and a gallery with all the icons, filterable by their name.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
